core/src/main.cpp

```diff
@@ -257,12 +496,23 @@ configuration options.
 --dump-config-template can be used to generate a default configuration file.
 See this option's description for more information.
 
+--dump-jsonschema can be used to generate a default schema file.
+See this option's description for more information.
+
 Options:
       --dump-config-template
                      Print configuration template to stdout and exit. Some
                      options have values which act as placeholders. If used
                      to generate a configuration file, those options will
                      need to be updated.
+      --dump-jsonschema
+                     Print default JSON schema to stdout and exit. The
+                     schema is used to validate the configuration file.
+      --jsonschema-file [SCHEMA_FILE_PATH]
+                     Validate server configuration against SCHEMA_FILE_PATH.
+                     Validation is performed before startup. If validation
+                     fails, the server will exit. If SCHEMA_FILE_PATH is not
+                     defined, the default is used.
```

Comment:
Post discussion notes:
- Always validate the config
- Make the schema file path required

Reply:
Done